ci/do_ci.sh

```diff
@@ -285,10 +285,7 @@ case $CI_TARGET in
         echo "Building API..."
         bazel build "${BAZEL_BUILD_OPTIONS[@]}" \
               -c fastbuild @envoy_api//envoy/...
-        if [[ -n "$ENVOY_API_ONLY" ]]; then
-            exit 0
-        fi
-        ;&
+        ;;
```

Comment:
`;&` is not supported by the bash of macos. And I will prefer to let the caller to handle the fall through by `./ci/do_ci.sh api && ./ci/do_ci.sh api.go`

I have no strong point to this. Any other suggestion or alternative solution would be great.

Reply:
cc @phlax

Reply:
or merge `api` and `api.go` to one case and handle it by a if check, for example:

```
    api|api.go)
        if [[ "${CI_TARGET}" == "api" ]]; then
            # Use libstdc++ because the API booster links to prebuilt libclang*/libLLVM* installed in /opt/llvm/lib,
            # which is built with libstdc++. Using libstdc++ for whole of the API CI job to avoid unnecessary rebuild.
            ENVOY_STDLIB="libstdc++"
            setup_clang_toolchain
            export CLANG_TOOLCHAIN_SETUP=1
            export LLVM_CONFIG="${LLVM_ROOT}"/bin/llvm-config
            echo "Run protoxform test"
            bazel run "${BAZEL_BUILD_OPTIONS[@]}" \
                --//tools/api_proto_plugin:default_type_db_target=//tools/testdata/protoxform:fix_protos \
                --//tools/api_proto_plugin:extra_args=api_version:3.7 \
                //tools/protoprint:protoprint_test
            echo "Validating API structure..."
            "${ENVOY_SRCDIR}"/tools/api/validate_structure.py
            echo "Testing API..."
            bazel_with_collection \
                test "${BAZEL_BUILD_OPTIONS[@]}" \
                --remote_download_minimal \
                -c fastbuild \
                @envoy_api//test/... \
                @envoy_api//tools/... \
                @envoy_api//tools:tap2pcap_test
            echo "Building API..."
            bazel build "${BAZEL_BUILD_OPTIONS[@]}" \
                -c fastbuild @envoy_api//envoy/...
            if [[ -n "$ENVOY_API_ONLY" ]]; then
                exit 0
            fi
        fi

        if [[ -z "$CLANG_TOOLCHAIN_SETUP" ]]; then
            setup_clang_toolchain
        fi
        GO_IMPORT_BASE="github.com/envoyproxy/go-control-plane"
        GO_TARGETS=(@envoy_api//...)
        read -r -a GO_PROTOS <<< "$(bazel query "${BAZEL_GLOBAL_OPTIONS[@]}" "kind('go_proto_library', ${GO_TARGETS[*]})" | tr '\n' ' ')"
        echo "${GO_PROTOS[@]}" | grep -q envoy_api || echo "No go proto targets found"
        bazel build "${BAZEL_BUILD_OPTIONS[@]}" \
              --experimental_proto_descriptor_sets_include_source_info \
              --remote_download_outputs=all \
              "${GO_PROTOS[@]}"
        rm -rf build_go
        mkdir -p build_go
        echo "Copying go protos -> build_go"
        BAZEL_BIN="$(bazel info "${BAZEL_BUILD_OPTIONS[@]}" bazel-bin)"
        for GO_PROTO in "${GO_PROTOS[@]}"; do
             # strip @envoy_api//
            RULE_DIR="$(echo "${GO_PROTO:12}" | cut -d: -f1)"
            PROTO="$(echo "${GO_PROTO:12}" | cut -d: -f2)"
            INPUT_DIR="${BAZEL_BIN}/external/envoy_api/${RULE_DIR}/${PROTO}_/${GO_IMPORT_BASE}/${RULE_DIR}"
            OUTPUT_DIR="build_go/${RULE_DIR}"
            mkdir -p "$OUTPUT_DIR"
            if [[ ! -e "$INPUT_DIR" ]]; then
                echo "Unable to find input ${INPUT_DIR}" >&2
                exit 1
            fi
            # echo "Copying go files ${INPUT_DIR} -> ${OUTPUT_DIR}"
            while read -r GO_FILE; do
                cp -a "$GO_FILE" "$OUTPUT_DIR"
                if [[ "$GO_FILE" = *.validate.go ]]; then
                    sed -i '1s;^;//go:build !disable_pgv\n;' "$OUTPUT_DIR/$(basename "$GO_FILE")"
                fi
            done <<< "$(find "$INPUT_DIR" -name "*.go")"
        done
        ;;

```

Reply:
i think we want to keep the separate targets - `api.go` is called by go-control-plane currently and i think they _are_ separate ci targets

Reply:
i think moving the api and api.go targets into functions is a good idea anyway - but it would allow something like

```bash
            build_api
            if [[ -z "$ENVOY_API_ONLY" ]]; then
                 build_api_go
            fi
```

... and the api.go can just call the latter, which is much cleaner i think

Reply:
it changes readability

Reply:
if we move the code itself out to functions - im more amenable to merging the targets (although think its not correct tbh) - but without doing that imho its the worst of all worlds

Reply:
okay, then let's do a quick update. My only target is to resolve the fall through.

Reply:
understood - its come up before and was never resolved - it was done this way for a reason tho (to separate out the code)